packages/protocol/engine/eviction/state.go

```diff
@@ -76,6 +78,21 @@ func (s *State) LastEvictedSlot() slot.Index {
 	return s.lastEvictedSlot
 }
 
+// EarliestRootCommitment returns the earliest commitment across all rootblocks.
+func (s *State) EarliestRootCommitment() (earliestCommitment commitment.ID) {
+	s.rootBlocks.ForEach(func(index slot.Index, storage *shrinkingmap.ShrinkingMap[models.BlockID, commitment.ID]) {
+		storage.ForEach(func(id models.BlockID, commitmentID commitment.ID) bool {
+			if commitmentID.Index() < earliestCommitment.Index() {
```

Comment:
Isn't this always false? `earliestCommitment.Index()` is the empty value `0` and any `commitmentID.Index()` is at least `0` as well